Bug fix: map to list in lines 170:173

fgypas opened the pull request:
## Description


Fixes #16 

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation updated

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

meric466 committed:
map to list in lines 170:173

Exon start/end need to be converted to list to be used in the rest of the code, so that positions variable in line 260 is not empty.

AngryMaciek commented:
Please remove `/.DS_Store` from the repo, these are macOS system files.

Commit:
Downgrade pandas minimum version requirement to work with pip

Commit:
Black format

Commit:
Increase --max-line-length to 120 in flake8 tests

Commit:
Update CI

Commit:
Update test file

fgypas committed:
Remove mac test from ci. Bump version to 0.4.1

fgypas commented:
The documentation can be updated after the docker is available in biocontainers (because we don't know the tag it will get).

AngryMaciek commented on .test/calculate-tin/sample.tsv:
```diff
@@ -1,40 +1,40 @@
 transcript	sample_name
```
This is the expected output, right?

All of the non-zero values changed(!) It seems that the methods is calculating now sth different then before... How do we know it is still correct?

fgypas replied:
Well, I am not sure about this...

fgypas replied:
@mzavolan So the issue here is that the old version of the tool and the new version give us different TIN scores. Can this be explained by the changes you and @meric466 introduced today?

AngryMaciek replied:
We generally observe that the scores were lower previously and are higher now. Given that we operate on a dummy `bam` file with not very much reads alligned I would tend to say that I expect the transcript coverage to be rather poor and thus - TIN score low. Much more in the range 0-20 then 20+ ... So I am leaning towards saying that the previous scores made more sense.

meric466 replied:
I will check some mouse results I generated a couple of months ago, and I will rerun them with the new script and see how they look.

AngryMaciek replied:
And it seems I was wrong:  
TIN measures uniformity of the coverage, not the coverage _per se_, therefore my argument above might not be valid...

AngryMaciek replied:
The problem was diagnosed and resolved by @meric466

AngryMaciek commented on requirements.txt:
```diff
@@ -1,7 +1,7 @@
 guppy3>=3.1.0
 matplotlib>=3.3.4
-pandas>=1.2.3
+pandas>=1.1
```
But why was this necessary? There were no issues with `pip` installation previously (?)

fgypas replied:
Yes, I run into this, but I was not upgrading pip. I will reverse this.

fgypas replied:
OK, done. Works fine.

AngryMaciek commented on setup.py:
```diff
@@ -9,7 +9,7 @@
 setup(
     name='tin-score-calculation',
-    version='0.4',
+    version='0.4.1',
```
If the method is calculating something different now (regardless if it was correct then or now) I think we should bump up major ver. (?)

fgypas replied:
Ok, 0.5 then?

AngryMaciek replied:
OK `0.5.0` is fine for me, since we don't have `1.0.0` yet. This would be a minor version bump then.

fgypas replied:
OK, done.

AngryMaciek reviewed (commented):
Apart from my other comments:
* I assume that changes to the `.py` scripts (**apart from** the fix by @meric466 ) were introduced by `black` only, so I do not check them.
* https://github.com/zavolanlab/tin-score-calculation/pull/17#issuecomment-879880443

fgypas committed:
Reverse pandas version. Make major release instead.

meric466 committed:
bug fix map to list comprehesion

We changed lines 168 to 171 and changed map function with list comprehension and tested the behavior with a dataset and the code worked fine.

AngryMaciek commented:
@meric466
Please make sure all the tests in the _CI_ pass, only then I can merge.
Also, take a look at the review comments here: https://github.com/zavolanlab/tin-score-calculation/pull/17#pullrequestreview-706390992

Commit:
Black formatting

fgypas commented:
> @meric466
> Please make sure all the tests in the _CI_ pass, only then I can merge.
> Also, take a look at the review comments here: [#17 (review)](https://github.com/zavolanlab/tin-score-calculation/pull/17#pullrequestreview-706390992)

@AngryMaciek @meric466 I just fixed the formatting issue, so the tests now pass.

AngryMaciek commented:
Thanks a lot! But https://github.com/zavolanlab/tin-score-calculation/pull/17#issuecomment-879880443 still stands.

meric466 committed:
DS_Store removed

DS_Store removed

meric466 commented:
> Thanks a lot! But [#17 (comment)](https://github.com/zavolanlab/tin-score-calculation/pull/17#issuecomment-879880443) still stands.

Thanks @fgypas , @AngryMaciek I removed the file, is it OK now?
